src/data/list/range.lean

```diff
@@ -148,6 +148,29 @@ by simp only [range_eq_range', range'_concat, zero_add]
 
 @[simp] lemma range_zero : range 0 = [] := rfl
 
+theorem chain_range_succ (r : ℕ → ℕ → Prop) (n a : ℕ) :
+  chain r a (range n.succ) ↔ r a 0 ∧ ∀ m < n, r m m.succ :=
+begin
+  rw range_succ,
+  induction n with n hn,
+  { simp },
+  { rw range_succ,
+    simp only [append_assoc, singleton_append, chain_append_cons_cons, chain.nil, and_true],
+    rw [hn, and_assoc, and.congr_right_iff],
+    exact λ _, forall_lt_succ.symm }
+end
+
+theorem chain'_range_succ (r : ℕ → ℕ → Prop) (n : ℕ) :
```

Comment:
Can't you prove this one in terms of `chain_range_succ`?

Reply:
I don't believe so, but now that I think about it, I might be able to do the opposite. Let me try.

Reply:
Yep, there goes, `chain_range_succ` is now a two-liner.